Source/lighting.cpp

```diff
@@ -797,145 +671,92 @@ void __fastcall DoUnVision(int nXPos, int nYPos, int nRadius)
 	}
 }
 
-void __fastcall DoVision(int nXPos, int nYPos, int nRadius, unsigned char doautomap, int visible)
+void __fastcall DoVision(int nXPos, int nYPos, int nRadius, BOOL doautomap, BOOL visible)
 {
-	char *v5;           // esi
-	int v6;             // esi
-	int v7;             // edi
-	unsigned char *v8;  // eax
-	int v9;             // ebx
-	int v10;            // ecx
-	unsigned char v11;  // dl
-	int v12;            // ecx
-	int v13;            // ecx
-	unsigned char v14;  // cl
-	unsigned char v15;  // dl
-	int v16;            // ecx
-	int v17;            // ecx
-	int i;              // [esp+Ch] [ebp-34h]
-	unsigned char *v19; // [esp+10h] [ebp-30h]
-	int v20;            // [esp+14h] [ebp-2Ch]
-	int v21;            // [esp+18h] [ebp-28h]
-	int v22;            // [esp+1Ch] [ebp-24h]
-	signed int v23;     // [esp+20h] [ebp-20h]
-	signed int v24;     // [esp+24h] [ebp-1Ch]
-	signed int v25;     // [esp+28h] [ebp-18h]
-	signed int v26;     // [esp+2Ch] [ebp-14h]
-	signed int v27;     // [esp+30h] [ebp-10h]
-	int v28;            // [esp+34h] [ebp-Ch]
-	int v29;            // [esp+38h] [ebp-8h]
-	unsigned char v30;  // [esp+3Fh] [ebp-1h]
-	unsigned char v31;  // [esp+3Fh] [ebp-1h]
-
-	v28 = nYPos;
-	v29 = nXPos;
-	if (nXPos >= 0 && nXPos <= MAXDUNX && nYPos >= 0 && nYPos <= MAXDUNY) {
-		if (doautomap) {
-			v5 = &dFlags[nXPos][nYPos];
-			if (*v5 >= 0) {
+	int nCrawlX, nCrawlY, nLineLen, nBlockerFlag, nTrans;
+	int j, k, v, x1adj, x2adj, y1adj, y2adj;
+
+	if(nXPos >= 0 && nXPos <= MAXDUNX && nYPos >= 0 && nYPos <= MAXDUNY) {
+		if(doautomap) {
+			if(dFlags[nXPos][nYPos] >= 0) {
 				SetAutomapView(nXPos, nXPos);
-				nYPos = v28;
-				nXPos = v29;
 			}
-			*v5 |= DFLAG_EXPLORED;
+			dFlags[nXPos][nYPos] |= 0x80;

```

Comment:
Do we have enums for these?

Reply:
We do, but I haven't checked to see if they're correct. At least according to the debug release, we have one:
`BFLAG_MONSTLR     = 0x10`
The `LR` seems to be connected with `sync.cpp` which has something named `sgwLRU`. Though I haven't figured it out yet.

Reply:
LR probably stands for light radius (no need to thank me) :D

Reply:
Done.